Main.elm

```diff
@@ -28,14 +28,22 @@ update msg model =
       in
         Model newHopAdditions (model.nextHopAdditionId + 1)
     UpdateHopAddition id hopAdditionMsg ->
-      let
-        updateHopAddition (hopAdditionID, hopAdditionModel) =
-          if hopAdditionID == id then
-            (hopAdditionID, HopAddition.update hopAdditionMsg hopAdditionModel)
-          else
-            (hopAdditionID, hopAdditionModel)
-      in
-        { model | hopAdditions = List.map updateHopAddition model.hopAdditions }
+      case hopAdditionMsg of
+        -- TODO: The fact that the parent knows about this child action makes me sad
```

Comment:
See code comment - I hate that the parent knows about the child action.

Is there a better way to handle this case? The delete button is in the child component, but the action has to remove the entire child model from the parent's list of child models.